doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,597 @@
+nonlinearity). This is applicable to any type of input, be it an image, a sound
+clip or an unordered collection of features: their representation can always be
+flattened into a vector before the transformation.
+
+In the case of images and sound clips, there is some ordering (spatial or
+temporal) that is thrown away when they are flattened. This information may
+prove very handy in solving some tasks, like computer vision and speech
+recognition.
+
+These sorts of inputs share important properties:
+
+* They are stored as N-dimensional arrays (called *tensors* in Theano
+  terminology).
+* They feature one or more axes for which ordering matters (e.g. width and
+  height axes for an image, time axis for a sound clip).
+* One axis, the channel axis, is used to access different views of the data
+  (e.g. the red, green and blue channels of a color image, or the left and
+  right channels of a stereo audio track).
+
+A discrete convolution is a linear transformation that has the following
+properties:
+
+* It is sparse, i.e. only a few input units contribute to a given output unit.
+* Parameters are shared, i.e. the same transformation is applied to multiple
+  locations in the input.
+
+Here is an example of a discrete convolution:
+
+.. figure:: conv_arithmetic_figures/numerical_no_padding_no_strides.gif
+
+The blue grid is called the input *feature map*. We slide a *kernel* (dark blue)
+with value
+
+.. math::
+
+    \begin{pmatrix}
+    0 & 1 & 2 \\
+    2 & 2 & 0 \\
+    0 & 1 & 2
+    \end{pmatrix}
+
+across the input feature map. At each location, we compute the product between
+the kernel elements and the input elements they overlap and sum the results up
+to obtain the corresponding output *feature map* value.
+
+.. note::
+
+    What we referred to earlier as *channels* is a name for the *feature maps*
+    of observed data.
+
+If there are multiple input feature maps, each input feature map is convolved
+with a distinct kernel, and the resulting feature maps are summed up elementwise
+to produce the output feature map. The procedure can be repeated using different
+kernels to form as many output feature maps as desired.
+
+The collection of kernels defining a discrete convolution has a shape
+corresponding to some permutation of ``(num_output_feature_maps,
+num_input_feature_maps, kernel_height, kernel_width)``.
+
+.. note::
+
+    While there is a distinction between convolution and cross-correlation from
+    a signal processing perspective, the two become interchangeable when the
+    kernel is learned. For the sake of simplicity and consistency, we will refer
+    to this operation as a *convolution* and ignore the implementation details.
+
+The convolution above is an instance of a 2-D convolution, but it can be
+generalized to N-D convolutions. For instance, in a 3-D convolution, the kernel
+would be a *cuboid* and would slide across the height, width and depth of the
+input feature map.
```

Comment:
This paragraph can be removed if we introduce the 3D/4D *cuboid* / *hypercuboid* in the previous description (see prev comment).